DownFall/Views/BackpackView.swift

```diff
@@ -16,79 +16,226 @@ import SpriteKit
+    private var viewContainer: SKSpriteNode
+    
+    private var inventoryArea: SKSpriteNode
+    
+    private var itemArea: SKSpriteNode?
+    private var descriptionArea: SKSpriteNode?
+    private var ctaButton: Button?
+    private var cancelButton: Button?
+    private var nextItemsButton: Button?
+    private var abilities: [AnyAbility] = []
+    
+    let targetingAreaName = "targetingArea"
+    
     var selectedAbility: AnyAbility? = nil {
         didSet {
+            viewModel.ability = selectedAbility
             updateDescription(with: selectedAbility)
             updateCTAButton(with: selectedAbility)
         }
     }
     
     var height: CGFloat = 0.0
     
-    init(playableRect: CGRect) {
+    init(playableRect: CGRect, viewModel: TargetingViewModel, levelSize: Int) {
         self.playableRect = playableRect
+        self.boardSize = CGFloat(levelSize)
         //height and width set ups
         height = playableRect.height * Style.Backpack.heightCoefficient
-        let backPackHeight: CGFloat = 280
-        let backPackWidth = height * 2/3
+        
+        // get the view model
+        self.viewModel = viewModel
+        
+        //get the tile size
+        self.tileSize = 0.9 * (playableRect.width / CGFloat(levelSize))
+        
+        // view container
+        self.viewContainer = SKSpriteNode(texture: nil, color: .clear, size: playableRect.size)
+        self.viewContainer.zPosition = Precedence.foreground.rawValue
+        
+        // background view
+        self.background = SKSpriteNode(color: .foregroundBlue, size: CGSize(width: playableRect.width, height: height))//setting the background position
+        self.background.position = CGPoint.positionThis(background.frame, inBottomOf: self.viewContainer.frame)
+        
+        //targeting area
+        self.targetingArea = SKSpriteNode(color: .clear, size: CGSize(width: playableRect.width, height: playableRect.height))
+        self.targetingArea.position = self.viewContainer.frame.center
+        self.targetingArea.name = self.targetingAreaName
+        
+        //inventory area
+        self.inventoryArea = SKSpriteNode(texture: nil, color: .clear, size: CGSize(width: playableRect.width, height: height))
+        self.inventoryArea.position = CGPoint.positionThis(self.inventoryArea.frame, inBottomOf: self.viewContainer.frame)
+
+        // center target area reticles
+        let marginWidth = playableRect.width - CGFloat(tileSize * boardSize)
+        let marginHeight = playableRect.height - CGFloat(tileSize * boardSize)
+        let bottomLeftX = playableRect.minX + marginWidth/2 + tileSize/2
+        let bottomLeftY = playableRect.minY + marginHeight/2 + tileSize/2
+        self.bottomLeft = CGPoint(x: bottomLeftX, y: bottomLeftY)
         
         super.init(texture: nil, color: .clear, size: CGSize(width: playableRect.width, height: height))
+        // "bind" to to the view model
+        self.viewModel.updateCallback = self.updated
```

Comment:
This should be a weak reference to ourselves because we already have a strong reference to the viewModel